CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+
+## Developer environment
+
+By default, all k4e projects rely on Makefiles, where users can run all the
+workflows operations.
+
+To make it easier, all Makefiles implement a help section to see the actions you
+can run. Here is an example:
+
+```
+--> make help
+Usage:
+  make <target>
+
+General
+  help             Display this help.
+
+Development
+  manifests        Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
+  generate         Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+  fmt              Run go fmt against code.
+  vet              Run go vet against code.
+  test             Run tests.
+
+Build
+  build            Build manager binary.
+  run              Run a controller from your host.
+  docker-build     Build docker image with the manager.
+  docker-push      Push docker image with the manager.
+
+Deployment
+  install          Install CRDs into the K8s cluster specified in ~/.kube/config.
+  uninstall        Uninstall CRDs from the K8s cluster specified in ~/.kube/config.
+  deploy           Deploy controller to the K8s cluster specified in ~/.kube/config.
+  undeploy         Undeploy controller from the K8s cluster specified in ~/.kube/config.
+  controller-gen   Download controller-gen locally if necessary.
+  kustomize        Download kustomize locally if necessary.
+  bundle           Generate bundle manifests and metadata, then validate generated files.
+  bundle-build     Build the bundle image.
+  bundle-push      Push the bundle image.
+  opm              Download opm locally if necessary.
+  catalog-build    Build a catalog image.
+  catalog-push     Push a catalog image.
+```
+
+The core maintainers are developing this project using the latest Fedora and
+RHEL distributions.  If you're using any other distribution, maybe you can hit
+any not know issue.
+
+### Testing:
+
+All projects implement the target `make test`, where dependencies should be
+checked before running the test.
+
+For testing, Ginkgo and Gomega is the prefered flavour. You can find multiple
+examples in the current project. The test rules are the following:
+- Test Suite should be run on different packages.
+- Gomock is the prefered flavour for mocking.
+- Should avoid any Conditionals or logic.
+- Common code should be moved as helper functions.
+- Test structure should follow the following code-comments
+```
+  It("Can retrieve message correctly", func() {
+    // given
+    device := getDevice("foo")
+    edgeDeviceRepoMock.EXPECT().
+      Read(gomock.Any(), "foo", testNamespace).
+      Return(device, nil).
+      Times(1)
+
+    // when
+    res := handler.GetControlMessageForDevice(context.TODO(), params)
+
+    // then
+    Expect(res).To(Equal(operations.NewGetControlMessageForDeviceOK()))
+  })
+```
+Projects are continuously tested, and the main branch should constantly be
+working.
+
+## Pull Request process
```

Comment:
We should mention DCO (https://github.com/kubevirt/vm-import-operator/blob/master/CONTRIBUTING.md#contributor-compliance-with-developer-certificate-of-origin-dco)